vm/src/vm.rs

```diff
@@ -100,6 +100,15 @@ pub struct PySettings {
 
     /// sys.argv
     pub argv: Vec<String>,
+
+    pub syscall_functions: PySyscallFunctions,
```

Comment:
I'm a big fan of the [dependency injection pattern](https://en.wikipedia.org/wiki/Dependency_injection), but, in this case I feel this is not the right choice.

Instead, I would solve this issue with `#cfg` macro's in the time module. Rationale for solving is like that is that I can imagine the `PySycallFunctions` struct grow to an unwieldy size.

Reply:
The issue is that that would require adding `wasm-bindgen` as a dependency of `rustpython-vm`, which I wasn't sure about doing.

Reply:
In this case I would recommend adding the wasm-bindgen as a dependency to the rustpython-vm crate. You can make this a wasm target only dependency with dependencies based upon configuration options.

At some point it might make sense to split the vm crate into two crate. One core vm crate, with the bare minimal modules, and a stdlib crate, which contains all the hairy details dealing with platform specifics.